FreeRTOS_Sockets.c

```diff
@@ -1878,371 +1980,539 @@ void * vSocketClose( FreeRTOS_Socket_t * pxSocket )
-                         * sleeps. */
-                        case FREERTOS_SO_SET_SEMAPHORE:
-                           {
-                               pxSocket->pxUserSemaphore = *( ipPOINTER_CAST( SemaphoreHandle_t *, pvOptionValue ) );
-                           }
-                            xReturn = 0;
-                            break;
-                    #endif /* ipconfigSOCKET_HAS_USER_SEMAPHORE */
+            pxProps = ipPOINTER_CAST( const WinProperties_t *, pvOptionValue );
 
-                    #if ( ipconfigSOCKET_HAS_USER_WAKE_CALLBACK != 0 )
-                        case FREERTOS_SO_WAKEUP_CALLBACK:
+            xReturn = prvSockopt_so_buffer( pxSocket, FREERTOS_SO_SNDBUF, &( pxProps->lTxBufSize ) );
 
-                            /* Each socket can have a callback function that is executed
-                             * when there is an event the socket's owner might want to
-                             * process. */
-                            /* The type cast of the pointer expression "A" to type "B" removes const qualifier from the pointed to type. */
-                            pxSocket->pxUserWakeCallback = ipCAST_CONST_PTR_TO_CONST_TYPE_PTR( SocketWakeupCallback_t, pvOptionValue );
-                            xReturn = 0;
-                            break;
-                    #endif /* ipconfigSOCKET_HAS_USER_WAKE_CALLBACK */
+            if( xReturn != 0 )
+            {
+                break; /* will return an error. */
+            }
 
-                    case FREERTOS_SO_SET_LOW_HIGH_WATER:
-                       {
-                           const LowHighWater_t * pxLowHighWater = ipPOINTER_CAST( const LowHighWater_t *, pvOptionValue );
+            xReturn = prvSockopt_so_buffer( pxSocket, FREERTOS_SO_RCVBUF, &( pxProps->lRxBufSize ) );
 
-                           if( pxSocket->ucProtocol != ( uint8_t ) FREERTOS_IPPROTO_TCP )
-                           {
-                               /* It is not allowed to access 'pxSocket->u.xTCP'. */
-                               FreeRTOS_debug_printf( ( "FREERTOS_SO_SET_LOW_HIGH_WATER: wrong socket type\n" ) );
-                               break; /* will return -pdFREERTOS_ERRNO_EINVAL */
-                           }
+            if( xReturn != 0 )
+            {
+                break; /* will return an error. */
+            }
 
-                           if( ( pxLowHighWater->uxLittleSpace >= pxLowHighWater->uxEnoughSpace ) ||
-                               ( pxLowHighWater->uxEnoughSpace > pxSocket->u.xTCP.uxRxStreamSize ) )
-                           {
-                               /* Impossible values. */
-                               FreeRTOS_debug_printf( ( "FREERTOS_SO_SET_LOW_HIGH_WATER: bad values\n" ) );
-                               break; /* will return -pdFREERTOS_ERRNO_EINVAL */
-                           }
+            #if ( ipconfigUSE_TCP_WIN == 1 )
+                {
```

Comment:
Extra scope brackets are not necessary.

Reply:
Like I said, this is a FreeRTOS habit. Maybe the reason is to limit the scope of any local variable that is introduced between the ifdef's.